### PR TITLE
Feat/configure amount of matches

### PR DESCRIPTION
**Pull request by taltaub22**

### CODE REVIEW

File: client/app/pages/tournament-settings/tournament-settings.component.html [hunk 1]
```diff
@@ -10,27 +10,48 @@ <h3>Settings</h3>
...
+                                <br>
+                                This setting is used when you import a partial schedule file or a team list.
+                            </label>
 
-                    <h5>Delete Tournament Data</h5>
-                    <label>Press Delete to delete Teams and Matches. <br/> Note: You must first delete all scoring data.</label>
-                    <button class="button alert" data-open="model-delete"
-                            (click)="setDeleteModel(tournamentDataService)" [disabled]="!hasDataInDb">Delete
-                    </button>
+                            {{settings['practiceAmountOfMatches'].display}}
```
Comment (idanstark42) [inline client/app/pages/tournament-settings/tournament-settings.component.html]: Please keep note of the words you use. These are the number of **rounds**, not **matches** in each **stage**. Please use correct uniform naming.

File: server/logic/settings_logic.js [hunk 2]
```diff
@@ -8,11 +8,27 @@ const SETTING_COLLECTION_NAME = 'settings'
...
+    'practiceAmountOfMatches': 1,
+    'rankingAmountOfMatches': 3
   }
+
   db.connection().then(connection => {
-    connection.db().collection(SETTING_COLLECTION_NAME).findOne().then(response => {
-      if (!response) {
+    connection.db().collection(SETTING_COLLECTION_NAME).find({}).toArray().then(response => {
+      const promises = []
+      if (response.length > 0) {
```
Comment (idanstark42) [inline server/logic/settings_logic.js]: What was wrong with `findOne`?

File: server/logic/settings_logic.js [hunk 3]
```diff
@@ -8,11 +8,27 @@ const SETTING_COLLECTION_NAME = 'settings'
...
+        response = response[0]
+        for (const setting of Object.keys(defaultSettings)) {
```
Comment (idanstark42) [inline server/logic/settings_logic.js]: Does this work with `const`?

File: server/logic/settings_logic.js [hunk 4]
```diff
@@ -8,11 +8,27 @@ const SETTING_COLLECTION_NAME = 'settings'
...
+          if (!response[setting]) {
+            const toSet = {}
+            toSet[setting] = defaultSettings[setting]
+            promises.push(connection.db().collection(SETTING_COLLECTION_NAME).updateOne({}, {
+              $set: toSet
+            }))
+          }
+        }
+        return Promise.all(promises)
```
Comment (idanstark42) [inline server/logic/settings_logic.js]: `promises`, only gets pushed one promise in the code above. Maybe it should'nt be an array?

File: server/routers/matchTeam_router.js [hunk 5]
```diff
@@ -28,44 +31,46 @@ exports.getRouter = function () {
...
+      matchAmount: 1
+    },
+    {
+      stageName: 'ranking',
+      matchAmount: 3
+    }
+  ]
 
-  return matches
+  let practice = settings.getSetting('practiceAmountOfMatches')
```
Comment (idanstark42) [inline server/routers/matchTeam_router.js]: Please use more indicative names. Here, and in the next line.

Comment (idanstark42) [inline server/routers/matchTeam_router.js]: Please use more indicative names. Here, and in the next line.

File: server/logic/settings_logic.js (hunk 3 above)
Comment (taltaub22) [reply server/logic/settings_logic.js]: Yes it does

File: server/routers/matchTeam_router.js (hunk 5 above)
Comment (taltaub22) [reply server/routers/matchTeam_router.js]: Any idea for a better name?

Comment (idanstark42) [reply server/routers/matchTeam_router.js]: if you must, `practiceSettingsPromise`. I personally like to use these kind of promises inline in the call to `Promise.all`, this way they don't require a naming, which is always wierd with Promises.

File: server/logic/settings_logic.js (hunk 4 above)
Comment (taltaub22) [reply server/logic/settings_logic.js]: @idanstark42  Why only one? the push is inside the loop, for each setting that does not exists

File: client/app/pages/tournament-settings/tournament-settings.component.ts [hunk 6]
```diff
@@ -66,7 +76,7 @@ export class TournamentSettingsComponent implements OnInit {
         this.tournamentSettingsService.saveSetting(setting, this.settings[setting].value).subscribe(
             response => {
                 this.loading = false;
-                this.notification.success("Setting saved successfully")
+                this.notification.success(`${this.settings[setting].display} saved successfully`)
             },
             err => {
                 this.notification.error("Oh no! Something went wrong while trying to save...")
```
[the remark this answers is not in the document]
Comment (taltaub22) [reply client/app/pages/tournament-settings/tournament-settings.component.ts]: I can try. I just did not want to show any error to the user.

File: client/app/pages/tournament-settings/tournament-settings.component.ts [hunk 7]
```diff
@@ -66,19 +80,42 @@ export class TournamentSettingsComponent implements OnInit {
...
+                .subscribe(
+                    response => {
+                        this.notification.success(`Number of rounds saved successfully`)
+                    },
+                    error => {
+                        this.notification.error("Oh no! Something went wrong while trying to save number of rounds")
+                    }
+                )
+        } else {
+            this.notification.error(`Number of round per stage need to be between ${this.MIN_AMOUNT_OF_ROUNDS} and ${this.MAX_AMOUNT_OF_ROUNDS}`)
```
Comment (alanggreen) [inline client/app/pages/tournament-settings/tournament-settings.component.ts]: Replace "Number of round per stage need to be between" with
"Valid rounds per stage is between"

File: client/app/pages/tournament-settings/tournament-settings.component.ts (hunk 6 above)
[the remark this answers is not in the document]
Comment (taltaub22) [reply client/app/pages/tournament-settings/tournament-settings.component.ts]: @alanggreen Every notification that the client is seeing, is sent to the server for logging.
The router also saves an error if it return an error to the client

[the remark this answers is not in the document]
Comment (alanggreen) [reply client/app/pages/tournament-settings/tournament-settings.component.ts]: But what about the reason for the error? The err should also go to the log

[the remark this answers is not in the document]
Comment (taltaub22) [reply client/app/pages/tournament-settings/tournament-settings.component.ts]: @alanggreen The error is logged on the server side

**Pull request merged**